openshift-gitops-operator/tasks/main.yml

```diff
@@ -0,0 +1,182 @@
+                  memory: 256Mi
+                requests:
+                  cpu: 250m
+                  memory: 128Mi
+            rbac:
+              defaultPolicy: ''
+              policy: |
+                g, system:cluster-admins, role:admin
+              scopes: '[groups]'
+            redis:
+              resources:
+                limits:
+                  cpu: 500m
+                  memory: 256Mi
+                requests:
+                  cpu: 250m
+                  memory: 128Mi
+            resourceExclusions: |
+              - apiGroups:
+                - tekton.dev
+                clusters:
+                - '*'
+                kinds:
+                - TaskRun
+                - PipelineRun
+            controller:
+              resources:
+                limits:
+                  cpu: 2000m
+                  memory: 2048Mi
+                requests:
+                  cpu: 250m
+                  memory: 1024Mi
+        state: present
+    - name: Wait until argocd is available
+      kubernetes.core.k8s_info:
+        name: argocd
+        namespace: test-gitops
+        api_version: argoproj.io/v1alpha1
+        kind: ArgoCD
+      register: argocd
+      until: argocd.resources[0].status.phase is defined and argocd.resources[0].status.phase == "Available"
+      retries: 10
+      delay: 10
+    # 2 - testing application
+    - name: Create argocd application
+      kubernetes.core.k8s:
+        name: app-spring-petclinic
+        namespace: openshift-gitops
+        api_version: argoproj.io/v1alpha1
+        kind: Application
+        definition:
+          metadata:
+            labels:
+              opdev: opcap
+          spec:
+            destination:
+              namespace: openshift-gitops
+              server: https://kubernetes.default.svc
+            project: default
+            source:
+              directory:
+                recurse: true
+              path: app
+              repoURL: https://github.com/redhat-developer/openshift-gitops-getting-started.git
+              targetRevision: main
+            syncPolicy:
+              syncOptions:
+                - CreateNamespace=true
+              automated:
+                prune: true
+                selfHeal: true
+          state: present
+    - name: Wait until application is synced
+      kubernetes.core.k8s_info:
+        name: app-spring-petclinic
+        namespace: openshift-gitops
+        api_version: argoproj.io/v1alpha1
+        kind: Application
+      register: app
+      until: app.resources[0].status.sync.status is defined and app.resources[0].status.sync.status == "Synced"
+      retries: 10
+      delay: 10
+  rescue:
+    - name: Print when failed
+      ansible.builtin.debug:
+        msg: "Error testing argocd with application. Err: {{ ansible_failed_result }}"
+    - name: Register failed tests
+      ansible.builtin.set_fact:
+        test_result: "failed"
+- name: Clean Up resources
```

Comment:
@acmenezes should we add a task to cleanup the `openshift-gitops` namespace? it sticks around, even when the operator is uninstalled.

Reply:
@madorn , second take on it. After investigation I can only conclude that something else wen't wrong when I tried to delete it. Now the code is deleting openshift-gitops and completing well all tasks. So I guess we may merge this one. Let me know what you think.